test/test_torch.py

```diff
@@ -8864,6 +8864,20 @@ def test_multidim(x, dim):
             x = torch.randn(dims, device=device)
             test_multidim(x, singleton_dim)
 
+            # check reducing median with NaNs
+            # If the element in the median is a NaN, there can be issues
+            # when comparining with other nan elements
+            if fn_name == 'median':
+                y = torch.full((1, 3), np.nan, dtype=torch.float64, device=device)
+                y[:, :1] = 1.1
+                values, indices = fn_tuple(y, dim=1)
+                expected_values = torch.tensor([nan], dtype=torch.float64, device=device)
+                # Due to some deterministic issues with indices order in the GPU with repeated values
+                # 0 and 1 indices are both valid.
+                expected_indices = torch.tensor([1], dtype=torch.int64, device=device)
+                self.assertEqual(values, expected_values)
+                self.assertLessEqual(indices, expected_indices)
```

Comment:
I don't really like this test, maybe check that y[index] is nan?

Reply:
sure!